src/main/java/sirius/kernel/commons/Amount.java

```diff
@@ -244,6 +244,26 @@ public BigDecimal getAmount() {
         return value;
     }
 
+    @Override
+    public int intValue() {
+        return value.intValue();
```

Comment:
Dont't we require null checks + fallback here e.g. for Amount.NOTHING?

Reply:
Hmm, to me it feels ok to have an NPE here when treating an `Amount.NOTHING` as `Number`, but I am open for discussion. For now, I have added such cases to the test, and I have reinforced the casting in `Value`

Reply:
Return Integer.MIN_VALUE - this should probably the most unexpected behaviour :-P

Reply:
One could return an IllegalStateException (?) as this would feel less "random" ;-)

Reply:
Right, that's actually a very valid point.

I quite like the idea of an exception, additionally we maybe could expand the filled check methods (if they currently don't have a code path for Amounts) to act like the Value is empty when the Amount is empty